homeassistant/components/recorder/statistics.py

```diff
@@ -340,11 +455,36 @@ def list_statistic_ids(
     ]
 
 
+def _statistics_during_period_query(
+    hass: HomeAssistant,
+    end_time: datetime | None,
+    statistic_ids: list[str] | None,
+    bakery: Any,
+    base_query: Iterable,
+    table: type[Statistics | StatisticsShortTerm],
+) -> Callable:
+    baked_query = hass.data[bakery](lambda session: session.query(*base_query))
+
+    baked_query += lambda q: q.filter(table.start >= bindparam("start_time"))
+
+    if end_time is not None:
+        baked_query += lambda q: q.filter(table.start < bindparam("end_time"))
+
+    if statistic_ids is not None:
+        baked_query += lambda q: q.filter(
+            table.metadata_id.in_(bindparam("metadata_ids"))
+        )
+
+    baked_query += lambda q: q.order_by(table.metadata_id, table.start)
+    return baked_query  # type: ignore[no-any-return]
+
+
 def statistics_during_period(
```

Comment:
Maybe this should be smarter:
 - If high-res (5-minute) statistics is requested but already pruned, fall back to low-res (hourly) statistics
 - If low-res statistics is requested, supplement with high-res statistics for the current hour

Reply:
We should not be smarter. Leave that up to the frontend. Otherwise the frontend always needs to check if smartness was applied!